### ISSUE TITLE
False positive error due to incorrect type inference of negative i64 reference

### ISSUE THREAD

**Issue by jeffreybolle** [intellij-rust/intellij-rust#10261]
## Environment

* **IntelliJ Rust plugin version:** 0.4.190.5263-223
* **Rust toolchain version:** 1.67.1 (d5a82bbd2 2023-02-07) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2022.3.3 (CL-223.8836.42)
* **Operating system:** Linux 5.19.0-35-generic
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.macros.proc.attr, org.rust.cargo.features.settings.gutter, org.rust.cargo.emulate.terminal, org.rust.macros.proc

## Problem description
In Rust `-x` where x is `&i64` is a `i64` and not a `&i64`.  IntelliJ Rust incorrectly inferences the type as `&i64` and put a red line under it if it is used where a `i64` is expected.

## Steps to reproduce

1. Create a new Rust project
2. Paste in the following code

```rust
fn foo(number: i64) {
    println!("{number}");
}

fn main() {
    let x = &1i64;
    foo(-x);
}
```
3. IntelliJ Rust puts a red line under `-x` with the following hover-over error: 

```
mismatched types [E0308] expected `i64`, found `&i64`
```

4. Running `cargo run` successfully builds and runs the code with the following output:

```
    Finished dev [unoptimized + debuginfo] target(s) in 0.00s
     Running `target/debug/example`
-1
```

## Screenshot
![image](https://user-images.githubusercontent.com/196781/225720270-bcd6ad5d-f262-4800-a6b0-ad6242919483.png)
